Java/measureeval/src/main/java/com/lantanagroup/link/measureeval/utils/CqlLogAppender.java

```diff
@@ -0,0 +1,93 @@
+
+public class CqlLogAppender extends AppenderBase<ILoggingEvent> {
+    private static final Logger logger = LoggerFactory.getLogger(CqlLogAppender.class);
+    private static final Pattern LOG_PATTERN = Pattern.compile("([\\w.]+)\\.(\\d+:\\d+-\\d+:\\d+)\\(\\d+\\):\\s*(\\{\\}|[^\\s]+)");
+
+    @Override
+    protected void append(ILoggingEvent event) {
+        String message = event.getFormattedMessage();
+        Matcher matcher = LOG_PATTERN.matcher(message);
+
+        if (matcher.find()) {
+            String libraryId = matcher.group(1);
+            String range = matcher.group(2);
+            String output = matcher.group(3)
+                    .replaceAll("org.hl7.fhir.r4.model.", "")
+                    .replaceAll("@[0-9A-Fa-f]{6,8}", "");
+            String cql = null;
+
+            // Group the resources in the output
+            output = groupResources(output);
+
+            MeasureEvaluatorCache measureEvalCache = ApplicationContextProvider.getApplicationContext().getBean(MeasureEvaluatorCache.class);
+            MeasureEvaluator evaluator = measureEvalCache.get(libraryId);       // Assume the measure id is the same as the library id since the log entry doesn't output the measure url/id
+
+            if (evaluator != null) {
+                cql = CqlUtils.getCql(evaluator.getBundle(), libraryId, range);
+            }
+
+            // Custom processing with libraryId and range
+            processLogEntry(libraryId, range, output, cql);
+        }
+    }
+
+    private String groupResources(String output) {
+        if (output == null || output.isEmpty() || output.equals("{}") || !output.startsWith("{")) {
+            return output;
+        }
+
+        output = output.substring(1, output.length() - 1);
+
+        if (output.endsWith(",")) {
+            output = output.substring(0, output.length() - 1);
+        }
+
+        String[] resources = output.split(",");
+        Map<String, Integer> resourceCount = new HashMap<>();
+
+        for (String resource : resources) {
+            resourceCount.put(resource, resourceCount.getOrDefault(resource, 0) + 1);
+        }
+
+        StringBuilder groupedOutput = new StringBuilder();
+        for (Map.Entry<String, Integer> entry : resourceCount.entrySet()) {
+            if (groupedOutput.length() > 0) {
+                groupedOutput.append(",");
+            }
+            groupedOutput.append(entry.getKey());
+            if (entry.getValue() > 1) {
+                groupedOutput.append("(").append(entry.getValue()).append(")");
+            }
+        }
+
+        return groupedOutput.toString();
+    }
+
+    private void processLogEntry(String libraryId, String range, String output, String cql) {
+        if (cql != null) {
+            Pattern definePattern = Pattern.compile("^define \"([^\"]+)\"");
+            Matcher matcher = cql != null ? definePattern.matcher(cql) : null;
+            if (matcher != null && matcher.find()) {
```

Comment:
Given the containing `if`, neither `cql` nor `matcher` can be null in this block.